dotnet/src/SemanticKernel.Core/Orchestration/KernelExecutionContextExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Extension methods for <see cref="IKernelExecutionContext"/>
+/// </summary>
+public static class KernelExecutionContextExtensions
+{
+    /// <summary>
+    /// Run a pipeline composed of synchronous and asynchronous functions.
+    /// </summary>
+    /// <param name="kernelContext">Kernel context</param>
+    /// <param name="input">Input</param>
+    /// <param name="pipeline">Target functions to run in sequence</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>Result of the function composition</returns>
+    public static Task<SKContext> RunAsync(this IKernelExecutionContext kernelContext, string input, ISKFunction[] pipeline, CancellationToken cancellationToken = default)
+    {
+        return kernelContext.RunAsync(new ContextVariables(input), pipeline, cancellationToken);
+    }
+
+    /// <summary>
+    /// Run a pipeline composed of synchronous and asynchronous functions.
+    /// </summary>
+    /// <param name="kernelContext">Kernel context</param>
+    /// <param name="input">Input</param>
+    /// <param name="pipeline">Target functions to run in sequence</param>
+    /// <returns>Result of the function composition</returns>
+    public static Task<SKContext> RunAsync(this IKernelExecutionContext kernelContext, string input, ISKFunction pipeline)
```

Comment:
Maybe better with the optional cancellationToken `CancellationToken cancellationToken = default`

Reply:
All overloads were removed.